src/main/kotlin/no/nav/syfo/service/VarselBusService.kt

```diff
@@ -0,0 +1,27 @@
+package no.nav.syfo.service
+
+import no.nav.syfo.kafka.dinesykmeldte.DineSykmeldteHendelseKafkaProducer
+import no.nav.syfo.kafka.varselbus.*
+import no.nav.syfo.kafka.varselbus.domain.EsyfovarselHendelse
+import no.nav.syfo.kafka.varselbus.domain.HendelseType.*
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+
+class VarselBusService(
+    val dineSykmeldteHendelseKafkaProducer: DineSykmeldteHendelseKafkaProducer,
+
+) {
+    private val log: Logger = LoggerFactory.getLogger("no.nav.syfo.service.VarselBusService")
+    private val oppfolgingsplanVarsler = OppfolgingsplanVarsler()
+    fun processVarselHendelse(varselHendelse: EsyfovarselHendelse) {
+        log.info("Sender varsel av type ${varselHendelse.type}")
+        when (varselHendelse.type) {
+            NL_OPPFOLGINGSPLAN_OPPRETTET -> dineSykmeldteHendelseKafkaProducer.sendHendelse(
+                    oppfolgingsplanVarsler.varselOpprettetOppfolgingsplanNL(varselHendelse)
+            )
+            NL_OPPFOLGINGSPLAN_SENDT_TIL_GODKJENNING -> dineSykmeldteHendelseKafkaProducer.sendHendelse(
+                oppfolgingsplanVarsler.varselSendtOppfolgingsplanNL(varselHendelse)
+            )
+        }
```

Comment:
Dette ble mye bedre, men jeg ville dratt det ennå lenger og laget en service man delegerer arbeidet til. Da slipper man at både `VarserlBusService` og `OppfolgingsplanVarsler` trenger å forholde seg til at det skal sendes til Dine sykmeldte. I tillegg skal det jo snart også sendes til arbeidsgivernotifikasjoner, så da hadde det blitt ennå mer kode her med det patternet.
```suggestion
        when (varselHendelse.type) {
            NL_OPPFOLGINGSPLAN_OPPRETTET -> oppfolgingsplanVarselService.sendOpprettetOppfolgingsplanNL(varselHendelse)
            NL_OPPFOLGINGSPLAN_SENDT_TIL_GODKJENNING -> oppfolgingsplanVarselService.sendSendtOppfolgingsplanNL(varselHendelse)
        }
```

Reply:
Tenker det er bedre med 1 enkelt when/switch, istedet for å delegere samme jobb / KafkaProducer kode inn i masse service-klasser som gjør det samme. Det kommer til å være flere varseltyper enn oppfølgingsplaner (e.g. DialogmoteVarsler), så da er det kanskje bedre å ha noen lettere funksjoner i VarselBusService eller en annen relatert klasse